File: lms-FE/src/features/signup/schemas.ts
import { z } from 'zod';

export const signupSchema = z.object({
    name: z.string().min(2, 'Name must be at least 2 characters'),
    email: z.string().email('Please enter a valid email address'),
    password: z.string().min(6, 'Password must be at least 6 characters'),
    requested_role: z.enum(['student', 'teacher'], {
        message: 'Please select a role',
    }),
});

export type SignupFormData = z.infer<typeof signupSchema>;

export interface SignupRequestRead {
    id: number;
    name: string;
    email: string;
    requested_role: string;
    approved_role: string | null;
    status: 'pending' | 'approved' | 'rejected';
    created_at: string;
    approved_at: string | null;
}

export interface PaginatedSignupRequests {
    items: SignupRequestRead[];
    total: number;
    page: number;
    size: number;
    pages: number;
}

export interface SignupApprovalRequest {
    approved_role?: 'student' | 'teacher' | null;
}

File: lms-BE/app/features/signup_requests/schemas.py
from pydantic import BaseModel, EmailStr, Field
from typing import Optional, Literal
from datetime import datetime


class SignupRequestCreate(BaseModel):
    name: str = Field(..., min_length=2, max_length=100)
    email: EmailStr
    password: str = Field(..., min_length=6)
    requested_role: Literal["student", "teacher", "principal"]
    school_id: Optional[int] = None


class SignupRequestRead(BaseModel):
    id: int
    name: str
    email: str
    requested_role: str
    approved_role: Optional[str] = None
    status: str
    school_id: Optional[int] = None
    school_name: Optional[str] = None
    created_at: datetime
    approved_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class SignupApprovalRequest(BaseModel):
    """Optional role override on approval. If omitted, uses requested_role."""
    approved_role: Optional[Literal["student", "teacher", "principal"]] = None


class PaginatedSignupRequests(BaseModel):
    items: list[SignupRequestRead]
    total: int
    page: int
    size: int
    pages: int
